### PR TITLE
Add support for setting sentry-native handler_strategy

### PR DESCRIPTION
**Pull request by markushi**
## :scroll: Description
Bump sentry-native, add options to SentryAndroidOptions, extend sentry-native init code.

## :bulb: Motivation and Context
https://github.com/getsentry/sentry-native/issues/1026

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Comment by markushi** [comment]
@bitsandfoxes do you know if the underlying .NET issue is solved by this? Or in other words: Should we move forward with this and get it merged?

**Comment by bitsandfoxes** [comment]
> @bitsandfoxes do you know if the underlying .NET issue is solved by this? Or in other words: Should we move forward with this and get it merged?

AFAIK this will resolve https://github.com/dotnet/android/issues/9055
I could [reproduce this locally](https://github.com/dotnet/android/issues/9055#issuecomment-2206579100)

**Comment by markushi** [comment]
@bitsandfoxes we eventually still want to have this merged, right?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
 
+## Unreleased
+
+- Add support for setting sentry-native handler_strategy ([#3671](https://github.com/getsentry/sentry-java/pull/3671))
+
+### Dependencies
+
+- Bump Native SDK from v0.7.8 to v0.7.16 ([#3671](https://github.com/getsentry/sentry-java/pull/3671))
+  - [changelog](https://github.com/getsentry/sentry-native/blob/master/CHANGELOG.md#0716)
+  - [diff](https://github.com/getsentry/sentry-native/compare/0.7.2...0.7.16)
```
Comment (romtsn) [inline CHANGELOG.md:11]:
```suggestion
  - [diff](https://github.com/getsentry/sentry-native/compare/0.7.8...0.7.16)
```

**Review by romtsn — APPROVED** [review]

**Pull request merged**